include/votca/xtp/cudapipeline.h

```diff
@@ -143,6 +149,38 @@ inline void CudaPipeline::gemm_internal(const M1 &A, const M2 &B, M3 &C,
   }
 }
 
+template <class M>
+inline void CudaPipeline::diag_gemm(const M &A, const CudaMatrix &b,
+                                    CudaMatrix &C) const {
+
+  if (b.cols() != 1 && b.rows() != 1) {
+    throw std::runtime_error("B Matrix in Cublas diag_gemm must be a vector");
+  }
+
+  cublasSideMode_t mode = CUBLAS_SIDE_RIGHT;
```

Comment:
So this function multiplies a general matrix A with a diagonal Matrix D. It can either execute A*D or D*A. The mode sets the respective operation

As D is symmetric we can also write this as A^T*D or A*D. So passing in A^T uses the D*A mode.